configuration/config-txt.md

```diff
@@ -612,6 +637,96 @@ It's generally a good idea to keep the core temperature below 70 degrees and the
 
 Most overclocking issues show up immediately with a failure to boot. If this occurs, hold down the `shift` key during the next boot which will temporarily disable all overclocking; this will allow you to boot successfully and then edit your settings.
 
+## Conditional Filters
+
+When a single SD card (or card image) is only being used with one Pi and one monitor, it's easy to simply set `config.txt` as required for that specific combination and keep it that way, amending only when something changes.
+
+However if one Pi is swapped between different monitors, or if the SD card (or card image) is being swapped between multiple Pis, a single set of settings may no longer be sufficient.  Conditional filters allow you to make certain sections of the config file used only in specific cases, allowing a single `config.txt` to create different configurations when read by different hardware.
+
+### The `[all]` filter
+
+This is the most basic filter: it resets all previously set filters and allows any settings listed below it to be processed on all hardware.
+
+    [all]
+
+It is usually a good idea to add an `[all]` filter at the end of groups of filtered settings to avoid unintentionally combining filters (see below).
+
+### The `[pi1]` and `[pi2]` filters
+
+Any settings below a `pi1` filter will only be applied to Pi1 (A, A+, B, B+) hardware.
+Any settings below a `pi2` filter will only be applied to Pi2 hardware.
+
+    [pi1]
+    [pi2]
+
+These are particularly useful for defining different `kernel`, `initramfs`, and `cmdline` settings, as the Pi1 and Pi2 require different kernels.  For example, to define separate initramfs images for each:
+
+    [pi1]
+    initramfs initrd.img-3.18.7+ followkernel
+    [pi2]
+    initramfs initrd.img-3.18.7-v7+ followkernel
+    [all]
+
+Remember to use the `[all]` filter at the end, so that any subsequent settings aren't limited to Pi2 hardware only.
+
+### The `[EDID=*]` filter
+
+When switching between multiple monitors while using a single SD card in your Pi, and where a blank config is not sufficient to automatically select the desired resolution for each one, this allows specific settings to be chosen based on the monitors' EDID names.
+
+To view the EDID name of a specific monitor, run the following command:
+
+    tvservice -n
+
+This will print something like this:
+
+    device_name=VSC-TD2220
+
+You can then specify settings that apply only to this monitor like so:
+
+    [EDID=VSC-TD2220]
+    hdmi_group=2
+    hdmi_mode=82
+    [all]
+
+This forces 1920x1080 DVT mode for this monitor, without affecting any other monitors.
+
+Note that these settings apply only at boot, so the monitor must be connected at boot time and the Pi must be able to read its EDID information to get the correct name.  Hotplugging a different monitor after boot will not reselect different settings.
+
+### The serial number filter
```

Comment:
I guess this is the only 'filter header' which doesn't use square brackets, but in this case there's probably nothing to suitable to put between square brackets anyway!